modules/quality/include/opencv2/quality/QualityPSNR.hpp

```diff
@@ -0,0 +1,117 @@
+// This file is part of OpenCV project.
+// It is subject to the license terms in the LICENSE file found in the top-level directory
+// of this distribution and at http://opencv.org/license.html.
+
+#ifndef OPENCV_QUALITY_QUALITYPSNR_HPP
+#define OPENCV_QUALITY_QUALITYPSNR_HPP
+
+#include <limits>   // numeric_limits
+#include "QualityBase.hpp"
+#include "QualityMSE.hpp"
+
+namespace cv {
+namespace quality {
+namespace detail {
+
+    // convert mse to psnr
+    inline CV_CONSTEXPR double mse_to_psnr( double mse, double max_pixel_value )
```

Comment:
> CV_CONSTEXPR

Why?

Reply:
for c++ `constexpr` support, in case the compiler sees an opportunity to take advantage of it.  I typically add this for small utility functions like this one.  Should it be removed?

Reply:
Does it work here? With "partial" specialization of const arguments (`max_pixel_value` is compile-time constant, but `mse` - doesn't)?

Reply:
@alalek you're right in that this particular library won't use it as a `constexpr` since `mse` won't be known at compile-time.  However, `mse` is known at compile time in the test app but I'm not concerned about test app performance.  Typically I'll mark something as `constexpr`  if it could potentially be used in such a context (like in this case), and leave it up to the caller/compiler to identify those situations as appropriate.  However, since you feel it is unnecessary in this context, I will remove it and add it again later if there is a legitimate use case in the module.